src/ui/suspense/util.ts

```diff
@@ -39,16 +39,22 @@ export function getAsyncParentFramesSuspense(
   pauseId: PauseId,
   asyncIndex: number
 ): Frame[] | undefined {
-  if (asyncIndex === 0) {
-    return getFramesSuspense(pauseId);
-  }
-  const asyncParentPauseId = getAsyncParentPauseIdSuspense(pauseId, asyncIndex);
-  if (!asyncParentPauseId) {
-    return;
-  }
-  const frames = getFramesSuspense(asyncParentPauseId);
-  if (!frames) {
-    return;
+  try {
+    if (asyncIndex === 0) {
+      return getFramesSuspense(pauseId);
+    }
+    const asyncParentPauseId = getAsyncParentPauseIdSuspense(pauseId, asyncIndex);
+    if (!asyncParentPauseId) {
+      return;
+    }
+    const frames = getFramesSuspense(asyncParentPauseId);
+    if (!frames) {
+      return;
+    }
+    return frames.slice(1);
+  } catch (e) {
+    // TODO [FE-795]: Communicate this to the user
+    console.error(e);
+    return [];
```

Comment:
Looks like this would break the whole _suspense_ part of this function (which relies on throwing a promise-like object to signal data being loaded).

At a minimum you'd probably want to do something like this in the `catch`:
```js
} catch (errorOrPromise) {
  if (isThennable(errorOrPromise)) {
    throw errorOrPromise;
  } else {
    // ...
  }
}
```

but truthfully, I'm initially uneasy about this change to swallow errors.

Reply:
It would probably be best to have some kind of `Result` data type for places where we expect fetching might fail. Otherwise it's difficult to distinguish between "Fetch succeeded, no results" and "Fetch errored", which often leads to the user being stranded as to what to do next. Curious to hear others thoughts on what that might look like though!

Reply:
Thanks for the feedback on the suspense side. That makes sense.

> but truthfully, I'm initially uneasy about this change to swallow errors.

Agree that it's far from ideal. This could be more surgical and only swallow the "too many points" error. Ultimately, this PR is trying to be a stopgap until we handle these error situations properly since this issue is blocking a customer.